src/objects/set.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2017-present PyO3 Project and Contributors
+//
+use std::{hash, collections};
+use ffi;
+use python::{Python, PythonObject};
+use conversion::ToPyObject;
+use objects::{PyObject, PyIterator};
+use err::{self, PyResult, PyErr};
+
+
+/// Represents a Python `set`
+pub struct PySet(PyObject);
+/// Represents a  Python `frozenset`
+pub struct PyFrozenSet(PyObject);
+
+pyobject_newtype!(PySet, PySet_Check, PySet_Type);
+pyobject_newtype!(PyFrozenSet, PyFrozenSet_Check, PyFrozenSet_Type);
+
+impl PySet {
+    /// Creates a new set.
+    ///
+    /// May panic when running out of memory.
+    pub fn new(py: Python, elements: &[PyObject]) -> PySet {
```

Comment:
What do you think if we change this to:

```rust
pub fn new<T: ToPythonObject>(py: Python: elements: &[T]) -> PySet {}
```

so that we can do:

```rust
let set = PySet::new(py, &[1, 2, 3]);
```

instead of

```rust
let set = PySet::new(py, &[1.to_py_object(py).into_object()]);
```